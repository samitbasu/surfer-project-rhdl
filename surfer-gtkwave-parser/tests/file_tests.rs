use std::path::PathBuf;

fn parse_file(file: &str) {
    let path = PathBuf::from("tests/files").join(file);
    let s = std::fs::read_to_string(path).unwrap();
    let (dirs, errs) = surfer_gtkwave_parser::Parser::new(&s).parse();
    let snapshot = format!(
        "## Input ##\n\n{}\n## Errors ##\n\n{}\n\n## Directives ##\n\n{}",
        s,
        errs.iter()
            .map(|e| format!("{e:?}"))
            .collect::<Vec<_>>()
            .join("\n"),
        dirs.iter()
            .map(|d| format!("{d:?}"))
            .collect::<Vec<_>>()
            .join("\n"),
    );
    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_suffix(file);
    settings.bind(|| {
        insta::assert_snapshot!(snapshot);
    });
}

#[test]
fn with_8_bit_everything() {
    parse_file("with_8_bit_everything.gtkw");
}

#[test]
fn with_8_bit_one() {
    parse_file("with_8_bit_one.gtkw");
}
