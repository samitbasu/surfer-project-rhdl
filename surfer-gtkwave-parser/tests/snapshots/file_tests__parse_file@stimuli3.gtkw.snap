---
source: surfer-gtkwave-parser/tests/file_tests.rs
expression: snapshot
---
## Input ##

[*]
[*] GTKWave Analyzer v3.3.100 (w)1999-2019 BSI
[*] Fri Jun 21 14:25:22 2024
[*]
[dumpfile] "C:\Users\Oscar\Desktop\stimuli3.vcd"
[dumpfile_mtime] "Wed Sep 13 14:58:03 2023"
[dumpfile_size] 152150
[savefile] "C:\Users\Oscar\Desktop\stimuli3.gtkw"
[timestart] 3022
[size] 1000 600
[pos] 119 45
*-4.681415 4630 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
[treeopen] testbench_unfold.
[sst_width] 266
[signals_width] 225
[sst_expanded] 1
[sst_vpaned_height] 153
@23
[color] 5
#{testbench_unfold.filter_test.outdata1[14:0]} testbench_unfold.filter_test.outdata1[14] testbench_unfold.filter_test.outdata1[13] testbench_unfold.filter_test.outdata1[12] testbench_unfold.filter_test.outdata1[11] testbench_unfold.filter_test.outdata1[10] testbench_unfold.filter_test.outdata1[9] testbench_unfold.filter_test.outdata1[8] testbench_unfold.filter_test.outdata1[7] testbench_unfold.filter_test.outdata1[6] testbench_unfold.filter_test.outdata1[5] testbench_unfold.filter_test.outdata1[4] testbench_unfold.filter_test.outdata1[3] testbench_unfold.filter_test.outdata1[2] testbench_unfold.filter_test.outdata1[1] testbench_unfold.filter_test.outdata1[0]
@28
testbench_unfold.filter_test.reset
@420
#{testbench_unfold.filter_test.outdata1[14:0]} testbench_unfold.filter_test.outdata1[14] testbench_unfold.filter_test.outdata1[13] testbench_unfold.filter_test.outdata1[12] testbench_unfold.filter_test.outdata1[11] testbench_unfold.filter_test.outdata1[10] testbench_unfold.filter_test.outdata1[9] testbench_unfold.filter_test.outdata1[8] testbench_unfold.filter_test.outdata1[7] testbench_unfold.filter_test.outdata1[6] testbench_unfold.filter_test.outdata1[5] testbench_unfold.filter_test.outdata1[4] testbench_unfold.filter_test.outdata1[3] testbench_unfold.filter_test.outdata1[2] testbench_unfold.filter_test.outdata1[1] testbench_unfold.filter_test.outdata1[0]
@200
-
@22
#{testbench_unfold.filter_test.outdata2[14:0]} testbench_unfold.filter_test.outdata2[14] testbench_unfold.filter_test.outdata2[13] testbench_unfold.filter_test.outdata2[12] testbench_unfold.filter_test.outdata2[11] testbench_unfold.filter_test.outdata2[10] testbench_unfold.filter_test.outdata2[9] testbench_unfold.filter_test.outdata2[8] testbench_unfold.filter_test.outdata2[7] testbench_unfold.filter_test.outdata2[6] testbench_unfold.filter_test.outdata2[5] testbench_unfold.filter_test.outdata2[4] testbench_unfold.filter_test.outdata2[3] testbench_unfold.filter_test.outdata2[2] testbench_unfold.filter_test.outdata2[1] testbench_unfold.filter_test.outdata2[0]
@200
-Foo
[pattern_trace] 1
[pattern_trace] 0

## Errors ##

UnknownDirective("[*]")
UnknownDirective("[*] GTKWave Analyzer v3.3.100 (w)1999-2019 BSI")
UnknownDirective("[*] Fri Jun 21 14:25:22 2024")
UnknownDirective("[*]")
UnknownDirective("[dumpfile_mtime] \"Wed Sep 13 14:58:03 2023\"")
UnknownDirective("[dumpfile_size] 152150")
UnknownDirective("[savefile] \"C:\\Users\\Oscar\\Desktop\\stimuli3.gtkw\"")
UnknownDirective("[timestart] 3022")
UnknownDirective("[size] 1000 600")
UnknownDirective("[pos] 119 45")
UnknownLine("*-4.681415 4630 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1")
UnknownDirective("[treeopen] testbench_unfold.")
UnknownDirective("[sst_width] 266")
UnknownDirective("[signals_width] 225")
UnknownDirective("[sst_expanded] 1")
UnknownDirective("[sst_vpaned_height] 153")
Other("didn't understand bit range in trace: 'color] '")
UnknownLine("#{testbench_unfold.filter_test.outdata1[14:0]} testbench_unfold.filter_test.outdata1[14] testbench_unfold.filter_test.outdata1[13] testbench_unfold.filter_test.outdata1[12] testbench_unfold.filter_test.outdata1[11] testbench_unfold.filter_test.outdata1[10] testbench_unfold.filter_test.outdata1[9] testbench_unfold.filter_test.outdata1[8] testbench_unfold.filter_test.outdata1[7] testbench_unfold.filter_test.outdata1[6] testbench_unfold.filter_test.outdata1[5] testbench_unfold.filter_test.outdata1[4] testbench_unfold.filter_test.outdata1[3] testbench_unfold.filter_test.outdata1[2] testbench_unfold.filter_test.outdata1[1] testbench_unfold.filter_test.outdata1[0]")
Other("couldn't parse end of bit range '14:0]} testbench_unfold.filter_test.outdata1[14] testbench_unfold.filter_test.outdata1[13] testbench_unfold.filter_test.outdata1[12] testbench_unfold.filter_test.outdata1[11] testbench_unfold.filter_test.outdata1[10] testbench_unfold.filter_test.outdata1[9] testbench_unfold.filter_test.outdata1[8] testbench_unfold.filter_test.outdata1[7] testbench_unfold.filter_test.outdata1[6] testbench_unfold.filter_test.outdata1[5] testbench_unfold.filter_test.outdata1[4] testbench_unfold.filter_test.outdata1[3] testbench_unfold.filter_test.outdata1[2] testbench_unfold.filter_test.outdata1[1] testbench_unfold.filter_test.outdata1[0': invalid digit found in string")
Other("couldn't parse end of bit range '14:0]} testbench_unfold.filter_test.outdata2[14] testbench_unfold.filter_test.outdata2[13] testbench_unfold.filter_test.outdata2[12] testbench_unfold.filter_test.outdata2[11] testbench_unfold.filter_test.outdata2[10] testbench_unfold.filter_test.outdata2[9] testbench_unfold.filter_test.outdata2[8] testbench_unfold.filter_test.outdata2[7] testbench_unfold.filter_test.outdata2[6] testbench_unfold.filter_test.outdata2[5] testbench_unfold.filter_test.outdata2[4] testbench_unfold.filter_test.outdata2[3] testbench_unfold.filter_test.outdata2[2] testbench_unfold.filter_test.outdata2[1] testbench_unfold.filter_test.outdata2[0': invalid digit found in string")
UnknownDirective("[pattern_trace] 1")
UnknownDirective("[pattern_trace] 0")

## Directives ##

Dumpfile("C:\\Users\\Oscar\\Desktop\\stimuli3.vcd")
Trace { path: "testbench_unfold.filter_test.reset", bit_range: None, flags: Some(Flags(bin | rjustify)) }
Trace { path: "-", bit_range: None, flags: Some(Flags(blank)) }
Trace { path: "-Foo", bit_range: None, flags: Some(Flags(blank)) }
